### PR TITLE
feat: contextElement for virtual elements

### PR DESCRIPTION
**Pull request by atomiks**
We didn't add `referenceNode` from Popper v1 because we thought it wasn't necessary anymore, but in some cases it still is. (Note Popper v1's implementation was bugged, this works correctly.)

Found via https://github.com/atomiks/tippyjs/issues/746#issuecomment-606275367

### CODE REVIEW

File: src/types.js [hunk 1]
```diff
@@ -135,4 +135,5 @@ export type Padding = number | $Shape<SideObject>;
 
 export type VirtualElement = {|
   getBoundingClientRect: () => ClientRect | DOMRect,
+  contextNode?: Element,
```
Comment (atomiks) [inline src/types.js]: `contextElement` instead maybe?

**Comment by atomiks** [comment]
CI tests aren't running anymore(?)

File: src/types.js (hunk 1 above)
Comment (FezVrasta) [reply src/types.js]: Yes I agree, since it expects an `Element`

**Comment by FezVrasta** [comment]
The build is here, but for some reason GitHub doesn't link it to the PR automatically.

https://travis-ci.org/github/popperjs/popper-core/builds/669135144

File: src/types.js (hunk 1 above)
Comment (FezVrasta) [reply src/types.js]: Should this be documented in the docs?

Comment (atomiks) [reply src/types.js]: We have to wait until after release. I'll make a docs PR for v2.2

**Pull request merged**